dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -781,60 +892,84 @@ ALWAYS_INLINE void Aggregator::executeImplBatch(
     }
 
     /// Generic case.
-
-    std::unique_ptr<AggregateDataPtr[]> places(new AggregateDataPtr[agg_size]);
+    std::unique_ptr<AggregateDataPtr[]> places(new AggregateDataPtr[rows]);
```

Comment:
can we also prefetch the `AggregateDataPtr`?

Reply:
+1. If accessing an entry in the hash table results in a cache miss, it is highly likely that accessing the AggregateDataPtr pointer within that entry will also result in a cache miss.

Reply:
done